helpers/student.js

```diff
@@ -95,13 +97,76 @@ exports.image = async function (req, res, next) {
     else {
       updatedFields = { profilePicURL: imageURL }
     }
-    const updatedStudent = await findAndUpdate(decodedToken.email, updatedFields);
-    res.send({ updatedStudent });
+    const updatedStudent = await studentModel.findOneAndUpdate({email: decoded.email}, updatedFields);
+    return res.status(200).json(updatedStudent)
   }
   catch (err) {
-    return res.json({ message: err.message });
+    return res.status(400).json({ message: err.message });
+  }
+}
+
+exports.getClubs = async function (req, res) {
+
+  // pull email from jwt
+  const token = req.headers.authorization.split(" ")[1];
+  const decoded = jwt.decode(token, { complete: true });
+  let id = decoded.payload.id;
+  console.log('Request made from:', id)
+
+  // Find which clubs the student follows
+  let clubs
+  try {
+    clubs = await studentModel.findOne({ email }, 'clubs');
+  }
```

Comment:
the purpose of refactoring email -> id was so we'd perform all queries with the id instead of the email. as you can see, `email` in line 119 is undefined, so most `find({ email })...`s need to be refactored to `find({ _id: id }), ...`. it's a small fix, but you'll have to do this in a fair few places!